client/app/layout/header/header.component.js

```diff
@@ -1,9 +1,18 @@
+function backgroundSelector () {
```

Comment:
For the class, we can set this via the angular lifecycle.  Please see me about this one and we can talk this out.